fgpyo/sam/__init__.py

```diff
@@ -590,6 +587,70 @@ def set_pair_info(r1: AlignedSegment, r2: AlignedSegment, proper_pair: bool = Tr
     r2.template_length = -insert_size
 
 
+@attr.s(auto_attribs=True, frozen=True)
+class ReadEditInfo:
+    """Counts various stats about how a read compares to a reference sequence."""
+
+    mismatches: int
+    insertions: int
+    inserted_bases: int
+    deletions: int
+    deleted_bases: int
+    nm: int
+
+
+def calc_edit_info(
+    r: AlignedSegment, reference_sequence: str, reference_offset: Optional[int] = None
+) -> ReadEditInfo:
+    """
+    Calculates the number of mismatches and the NM tag value for a read.  The read must be aligned.
+
+    Args:
+        r: the read/record for which to calculate values
+        reference_sequence: the reference sequence (or fragment thereof) that the read is
+          aligned to
+        reference_offset: if provided, assume that reference_sequence[reference_offset] is the
+          first base aligned to in reference_sequence, otherwise use r.reference_start
+
+    Returns:
+        a ReadEditInfo with information about how the read differs from the reference
+    """
+    query_offset = 0
+    target_offset = reference_offset if reference_offset is not None else r.reference_start
+    cigar = Cigar.from_cigartuples(r.cigartuples)
+
+    mms, insertions, ins_bases, deletions, del_bases = 0, 0, 0, 0, 0
+    ok_bases = ["A", "C", "G", "T"]
```

Comment:
make into a set, so we don't have to do a linear scan for containment below
```suggestion
    ok_bases = {"A", "C", "G", "T"}
```